src/TeaAsAService/Tea.Web/API/UserController.cs

```diff
@@ -20,6 +22,37 @@ public UserController(IDataStore dataStore)
             _dataStore = dataStore;
         }
 
+        [HttpPost]
+        [Route("createuser")]
+        public async Task<IActionResult> CreateUser([FromBody] CreateUserModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                var message = string.Join(" | ", ModelState.Values
+                    .SelectMany(v => v.Errors)
+                    .Select(e => e.ErrorMessage));
+                return BadRequest(message);
+            }
+
+
+            if(string.IsNullOrEmpty(model.LocalizedString))
+            {
+                 model.LocalizedString = HttpContext.Request.GetTypedHeaders()
+                .AcceptLanguage.OrderByDescending(x=>x.Quality ?? 0.1).FirstOrDefault()?.Value.ToString()
+                ?? "en-GB";
+            }         
+
+            if(!model.Password.ValidatePassword())

```

Comment:
And thinking about it, we could do the same with the email address as well. Make EmailAddress have only a getter on the Core.User, create a SetEmailAddress method on user like the SetPassword and just check to make sure the email address is valid.

Reply:
would I set the setter to private set? I'm not sure how to make the SetPassword method work with a readonly property. Would SetPassword be called in the CreateNewUser Method or in the UserController?

Reply:
Hey, yes do private set; on the property, it just means you will have to call user.setpassword to actually update the password property, you won't be able to do it outside of the class

Reply:
Haven't attempted the SetEmail yet. Wanted to make sure those first changes worked. For some reason the tests are not working on my machine, so I had to push the changes to check.

Reply:
Yes mate that look great. Not sure why the tests would fail to run locally they seem ok to me. Just double check the Mysql docker container is running, if it is then might be work just stopping it, the making sure the container has been removed in case its still got persisting an older version of the database or something.

Reply:
Just Finished up the email validation as well, so let me know if needs some tweaking.